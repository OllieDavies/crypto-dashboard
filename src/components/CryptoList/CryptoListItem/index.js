import React from "react";
import { Link } from "react-router-dom";

import ChangeIndicator from "../../../components/ChangeIndicator";
import styles from "./styles.module.css";

const CryptoListItem = ({ id, name, symbol, price, change }) => (
  <Link to={`./${id}`} className={styles.wrapper} data-testid="list-item-link">
    <div className={styles.container}>
      <div className={styles.nameContainer}>
        <h4 className={styles.name}>{name}</h4>
        <p className={styles.price}>
          {price} {symbol}
        </p>
      </div>
      <div className={styles.changeContainer}>
        <ChangeIndicator change={change} />
      </div>
    </div>
  </Link>
);

export default CryptoListItem;
