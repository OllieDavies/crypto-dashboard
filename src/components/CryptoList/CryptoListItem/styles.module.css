.wrapper {
  display: block;
  color: inherit;
  text-decoration: none;
  border: 1px black solid;
  border-radius: 8px;
  margin: 8px 0;
}

.container {
  display: flex;
  padding: 12px 16px;
}

.nameContainer {
  flex: 1;
}

.name {
  margin: 0 0 8px 0;
  font-size: 1.5em;
}

.price {
  margin: 0;
  color: rgba(0, 0, 0, 0.5);
}

.changeContainer {
  align-self: center;
}