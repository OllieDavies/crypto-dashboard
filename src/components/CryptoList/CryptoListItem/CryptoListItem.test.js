import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import CryptoListItem from ".";

describe("<CryptoListItem />", () => {
  const { container } = render(
    <MemoryRouter>
      <CryptoListItem
        id={1}
        name={"Bitcoin"}
        symbol={"BTC"}
        price={1000}
        change={0.1}
      />
    </MemoryRouter>
  );

  it("matches snapshot", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
