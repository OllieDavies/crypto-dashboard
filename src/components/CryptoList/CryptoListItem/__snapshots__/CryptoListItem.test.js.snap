// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CryptoListItem /> matches snapshot 1`] = `
<a
  class="wrapper"
  data-testid="list-item-link"
  href="/1"
>
  <div
    class="container"
  >
    <div
      class="nameContainer"
    >
      <h4
        class="name"
      >
        Bitcoin
      </h4>
      <p
        class="price"
      >
        1000
         
        BTC
      </p>
    </div>
    <div
      class="changeContainer"
    >
      <span
        class="text positive"
      >
        0.1
      </span>
    </div>
  </div>
</a>
`;
